versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapter.java

```diff
@@ -325,12 +322,17 @@ protected boolean doTryUpdateRepositoryDescription(
                   Filters.and(
                       Filters.eq(globalPointerKey), Filters.eq(DATA_PROPERTY_NAME, expectedBytes)),
                   doc);
-      return result.wasAcknowledged()
-          && result.getMatchedCount() == 1
-          && result.getModifiedCount() == 1;
+      return verifySuccessfulUpdate(result, client.getRepoDesc());
     } else {
-      InsertOneResult result = client.getRepoDesc().insertOne(doc);
-      return result.wasAcknowledged();
+      try {
+        return client.getRepoDesc().insertOne(doc).wasAcknowledged();
+      } catch (MongoWriteException writeException) {
+        ErrorCategory category = writeException.getError().getCategory();
+        if (ErrorCategory.DUPLICATE_KEY.equals(category)) {
```

Comment:
This change appears to go beyond code style scope :thinking:

Reply:
Yea